corehq/apps/translations/app_translations/upload_module.py

```diff
@@ -0,0 +1,291 @@
+# coding=utf-8
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import itertools
+import re
+from six.moves import zip
+
+from django.contrib import messages
+from django.utils.translation import ugettext as _
+
+from corehq import toggles
+from corehq.apps.app_manager.models import ReportModule
+from corehq.apps.translations.app_translations_old import get_unicode_dicts, _update_translation_dict
+
+
+
+def update_app_from_module_sheet(app, sheet):
+    """
+    Modify the translations of a module case list and detail display properties
+    given a sheet of translation data. The properties in the sheet must be in
+    the exact same order that they appear in the bulk app translation download.
+    This function does not save the modified app to the database.
+
+    :param sheet:
+    :param app:
+    :return:  Returns a list of message tuples. The first item in each tuple is
```

Comment:
repeating myself a lot but could be `namedtuple` here too. though also realize this was just moved.

Reply:
Leaving this one alone.